src/com/sap/piper/TemporaryCredentialsUtils.groovy

```diff
@@ -8,47 +8,63 @@ class TemporaryCredentialsUtils implements Serializable {
         this.script = script
     }
 
-    void handleTemporaryCredentials(List credentialItems, String credentialsDirectory, Closure body) {
+    void handleTemporaryCredentials(List credentialItems, List credentialsDirectories, Closure body) {
         final String credentialsFileName = 'credentials.json'
 
-        if (!credentialsDirectory) {
-            script.error("This should not happen: Directory for credentials file not specified.")
+        if (!credentialsDirectories) {
+            script.error("This should not happen: Directories for credentials files not specified.")
         }
 
         final boolean useCredentials = credentialItems
         try {
             if (useCredentials) {
-                writeCredentials(credentialItems, credentialsDirectory, credentialsFileName)
+                writeCredentials(credentialItems, credentialsDirectories, credentialsFileName)
             }
             body()
         }
         finally {
             if (useCredentials) {
-                deleteCredentials(credentialsDirectory, credentialsFileName)
+                deleteCredentials(credentialsDirectories, credentialsFileName)
             }
         }
     }
 
-    private void writeCredentials(List credentialItems, String credentialsDirectory, String credentialsFileName) {
+    private void writeCredentials(List credentialItems, List credentialsDirectories, String credentialsFileName) {
         if (!credentialItems) {
             script.echo "Not writing any credentials."
             return
         }
 
-        assertSystemsFileExists(credentialsDirectory)
+        Boolean systemsFileFound = false
+        for (int i = 0; i < credentialsDirectories.size(); i++) {
+            if (credentialsDirectories[i] && !credentialsDirectories[i].endsWith("/")) {
```

Comment:
Does is make sense to check for null (only) here? Not sure if line 43 or other lines would work for this case.

Reply:
```suggestion
            if (!credentialsDirectories[i]) {
                continue
            }
            if (!credentialsDirectories[i].endsWith("/")) {
```

Reply:
Actually I wasn't even thinking of the null check, but instead of passing an **empty** string instead of `./` to indicate the file resides in the root of the project. Hence if you implement it like you suggested it wouldn't work in case an empty string is provided, not sure if it should be like this.

Reply:
I guess it's better to do it as you suggested and have the separate check.